test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/ProjectSystems/LegacyPackageReferenceProjectTests.cs

```diff
@@ -96,6 +96,100 @@ public LegacyPackageReferenceProjectTests(DispatcherThreadFixture fixture)
         }
 
         [Fact]
+        public async Task GetCacheFilePathAsync_WithValidBaseIntermediateOutputPath_Succeeds()
+        {
+            // Arrange
+            using (var testDirectory = TestDirectory.Create())
+            {
+                var testProj = "project.csproj";
+                var testBaseIntermediateOutputPath = Path.Combine(testDirectory, "obj");
+                TestDirectory.Create(testBaseIntermediateOutputPath);
+                var projectAdapter = Mock.Of<IVsProjectAdapter>();
+                Mock.Get(projectAdapter)
+                    .SetupGet(x => x.BaseIntermediateOutputPath)
+                    .Returns(testBaseIntermediateOutputPath);
+
+                Mock.Get(projectAdapter)
+                    .SetupGet(x => x.FullProjectPath)
+                    .Returns(Path.Combine(testDirectory, testProj));
+
+                var testProject = new LegacyPackageReferenceProject(
+                    projectAdapter,
+                    Guid.NewGuid().ToString(),
+                    new TestProjectSystemServices(),
+                    _threadingService);
+
+                await _threadingService.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                // Act
+                var cachePath = await testProject.GetCacheFilePathAsync();
+
+                // Assert
+                Assert.Equal(Path.Combine(testBaseIntermediateOutputPath, $"{testProj}.nuget.cache"), cachePath);
+
+                // Verify
+                Mock.Get(projectAdapter)
+                    .VerifyGet(x => x.BaseIntermediateOutputPath, Times.AtLeastOnce);
+            }
+        }
+
+        [Fact]
+        public async Task GetCacheFilePathAsync_WithNoBaseIntermediateOutputPath_Throws()
```

Comment:
This will likely fail now that that you removed the thorws. 
I think it'll throw an NPE in path combine but the test being run the CI should clear it up :)